### ISSUE TITLE
Aws::Errors::MissingCredentialsError: unable to sign request without credentials set

### ISSUE THREAD

**Issue by nickooolas** [phstc/shoryuken#340]
Great work on launching V3.0.0 😁  @phstc - awesome to see we are on the concurrent_ruby now 👍 

I have read the release notes for the breaking changes, and understood I could change the existing initializer to something like follows:

`/initializers/shoryuken.rb`
```
Shoryuken.sqs_client = Aws::SQS::Client.new(secret_access_key: ENV['AWS_SQS_SECRET_ACCESS_KEY'],
                                            access_key_id:     ENV['AWS_SQS_ACCESS_KEY_ID'],
                                            region:            ENV['AWS_SQS_REGION'])
```
**NB: I did use different ENV variable names, not the default eg `AWS_REGION` I used `AWS_SQS_REGION`, as I like to keep these seperate access keys/secret keys for each service.**

I would have thought this would be sufficient to then run:

`bundle exec shoryuken -r ./app/workers/messages_inbound_sqs_worker -C ./config/shoryuken.yml -R -q awesome_queue_for_queue_messages`

But I'm getting:

```
bundler: failed to load command: shoryuken (/app/vendor/bundle/ruby/2.3.0/bin/shoryuken)
Aws::Errors::MissingCredentialsError: unable to sign request without credentials set
```

**Comment by nickooolas** [comment]
Some more info and debugging:

## Test 1 - Initialise an SQS client through `Shoyuken.sqs_client`, then try to use the Shoryuken.sqs_client (as I'd expect the worker does?) using Rails Console session, in Staging environment (Heroku Bash Session)
```
irb(main):001:0> Shoryuken.sqs_client = Aws::SQS::Client.new(secret_access_key: ENV['AWS_SQS_SECRET_ACCESS_KEY'],
irb(main):002:1*                                             access_key_id:     ENV['AWS_SQS_ACCESS_KEY_ID'],
irb(main):003:1*                                             region:            ENV['AWS_SQS_REGION'])
=> #<Aws::SQS::Client>
irb(main):004:0>
irb(main):005:0* Shoryuken.sqs_client.get_queue_attributes(queue_url: 'https://sqs.eu-west-2.amazonaws.com/0123456789/awesome_queue_for_queue_messages')
Aws::Errors::MissingCredentialsError: unable to sign request without credentials set
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/request_signer.rb:104:in `require_credentials'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/request_signer.rb:94:in `sign_authenticated_requests'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/request_signer.rb:87:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/helpful_socket_errors.rb:10:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/retry_errors.rb:88:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/sqs_queue_urls.rb:13:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/query/handler.rb:27:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/user_agent.rb:12:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/seahorse/client/plugins/endpoint.rb:41:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/param_validator.rb:21:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/idempotency_token.rb:18:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/seahorse/client/plugins/response_target.rb:21:in `call'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/seahorse/client/request.rb:70:in `send_request'
	from /app/vendor/bundle/ruby/2.3.0/gems/aws-sdk-core-2.8.7/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
	from (irb):5
	from /app/vendor/bundle/ruby/2.3.0/gems/railties-5.0.1/lib/rails/commands/console.rb:65:in `start'
	from /app/vendor/bundle/ruby/2.3.0/gems/railties-5.0.1/lib/rails/commands/console_helper.rb:9:in `start'
	from /app/vendor/bundle/ruby/2.3.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:78:in `console'
	from /app/vendor/bundle/ruby/2.3.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
	from /app/vendor/bundle/ruby/2.3.0/gems/railties-5.0.1/lib/rails/commands.rb:18:in `<top (required)>'
	from /app/bin/rails:9:in `require'
	from /app/bin/rails:9:in `<main>'
```
####Fails 😢 

## Test 2 - In the same Rails Console Session - initialise a normal SQS Client this time, with the same env variables, in the same Rails Console session and same Staging environment (Heroku Bash Session)
```
irb(main):006:0> client = Aws::SQS::Client.new(secret_access_key: ENV['AWS_SQS_SECRET_ACCESS_KEY'],
irb(main):007:1*                               access_key_id:     ENV['AWS_SQS_ACCESS_KEY_ID'],
irb(main):008:1*                               region:            ENV['AWS_SQS_REGION'])
=> #<Aws::SQS::Client>
irb(main):009:0> client.get_queue_attributes(queue_url: 'https://sqs.eu-west-2.amazonaws.com/0123456789/awesome_queue_for_queue_messages')
=> #<struct Aws::SQS::Types::GetQueueAttributesResult attributes={}>
irb(main):010:0>
```
###Success 🤦‍♂️

**Comment by phstc** [comment]
hi @nickooolas 

First of all, thanks for giving all details in the issue report 🍻 

> UPDATE - So only just saw your latest PR #335, all sorted and working now when using from master branch 🙌 , any ideas when you expect to be rolling into new Gem version?

True! @h3poteto fixed that 👏 👏 🍻 in #335. I've just released 3.0.2 with his fix.
